test/test.js

```diff
@@ -9,22 +9,61 @@ const uuid = require('node-uuid');
 
 describe('nodejs-docker', () => {
 
-  it('serves traffic on 8080', (done) => {
-    runDocker('test/express', 8080, (host, callback) => {
-      setTimeout(() => {
-        request(`http://${host}:8080`, (err, res, body) => {
-          if (err) {
-            console.error(`Error requesting content: ${util.inspect(err)}`);
-            throw err;
-          }
-          assert.equal(body, 'Hello World!');
-          callback(() => {
-            done();
+  function verifyOutput(description, dockerImage, expectedOutput) {
+    it(description, (done) => {
+      runDocker(dockerImage, 8080, (host, callback) => {
+        setTimeout(() => {
+          request(`http://${host}:8080`, (err, res, body) => {
+            try {
+              if (err) {
+                console.error(`Error requesting content: ${util.inspect(err)}`);
+                throw err;
+              }
+              assert.equal(body, expectedOutput);
+            }
+            finally {
+              callback(() => {
+                done();
```

Comment:
Does this ever fail? The `throw` above should be silenced by mocha since it is asynchronous and the test will always call done which should log the test as a success if I understand correctly.